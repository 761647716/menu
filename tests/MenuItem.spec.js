/* eslint-disable no-undef */
import React from 'react';
import { mount } from 'enzyme';
import KeyCode from 'rc-util/lib/KeyCode';
import Menu, { MenuItem } from '../src';

describe('MenuItem', () => {
  describe('disabled', () => {
    it('can not be active by key down', () => {
      const wrapper = mount(
        <Menu activeKey="1">
          <MenuItem key="1">1</MenuItem>
          <MenuItem disabled/>
          <MenuItem key="2">2</MenuItem>
        </Menu>
      );

      wrapper.simulate('keyDown', { keyCode: KeyCode.DOWN });
      expect(wrapper.find('MenuItem').at(1).props().active).toBe(false);
    });

    it('not fires select event when selected', () => {
      const handleSelect = jest.fn();
      const wrapper = mount(
        <Menu>
          <MenuItem disabled onSelect={handleSelect}>
            <span className="xx">Item content</span>
          </MenuItem>
        </Menu>
      );

      wrapper.find('.xx').simulate('click');
      expect(handleSelect).not.toBeCalled();
    });
  });

  describe('unmount', () => {
    it('removes self from selectedKeys', () => {
      const wrapper = mount(
        <Menu>
          <MenuItem key="1">1</MenuItem>
        </Menu>
      );
      wrapper.find('MenuItem').simulate('click');
      expect(wrapper.state('selectedKeys')).toEqual(['1']);
      wrapper.setProps({ children: null });
      expect(wrapper.state('selectedKeys')).toEqual([]);
    });
  });
});
