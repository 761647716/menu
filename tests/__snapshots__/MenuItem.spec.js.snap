// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem rest props can render all props to sub component 1`] = `
<ul
  aria-activedescendant=""
  class="rc-menu  rc-menu-root rc-menu-inline"
  role="menu"
  tabindex="0"
>
  <li
    aria-selected="false"
    class="rc-menu-item className rc-menu-item-active"
    data-whatever="whatever"
    role="menuitem"
    style="font-size: 20px; padding-left: 24px;"
    title="title"
  >
    1
  </li>
  <li
    class="rc-menu-submenu rc-menu-submenu-inline className"
    data-whatever="whatever"
    style="font-size: 20px;"
  >
    <div
      aria-expanded="false"
      aria-haspopup="true"
      aria-owns="item_1$Menu"
      class="rc-menu-submenu-title"
      style="padding-left: 24px;"
      title="title"
    >
      title
      <i
        class="rc-menu-submenu-arrow"
      />
    </div>
    <div />
  </li>
  <li
    class="className rc-menu-item-group"
    data-whatever="whatever"
    style="font-size: 20px;"
  >
    <div
      class="rc-menu-item-group-title"
      title="title"
    >
      title
    </div>
    <ul
      class="rc-menu-item-group-list"
    >
      <li
        aria-selected="false"
        class="rc-menu-item className"
        data-whatever="whatever"
        role="menuitem"
        style="font-size: 20px; padding-left: 24px;"
        title="title"
      >
        4
      </li>
    </ul>
  </li>
</ul>
`;
